packages/medusa/src/api/routes/admin/batch/get-batch-job.ts

```diff
@@ -0,0 +1,29 @@
+import { BatchJobService } from "../../../../services"
+/**
+ * @oas [get] /batch/{id}
+ * operationId: "GetBatchJobsBatchJob"
+ * summary: "Retrieve a Batch Job"
+ * description: "Retrieves a Batch Job."
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The id of the Batch Job
+ * tags:
+ *   - Batch Job
+ * responses:
+ *  "200":
+ *    description: OK
+ *    content:
+ *      application/json:
+ *        schema:
+ *          properties:
+ *            batch_job:
+ *              $ref: "#/components/schemas/batch_job"
+ */
+export default async (req, res) => {
+  const { id } = req.params
+
+  const batchJobService: BatchJobService = req.scope.resolve("batchJobService")
+
```

Comment:
comment: We should only allow the creator of the batch job to retrieve. Will create an issue for it.